### DISCUSSION TITLE
Environments variables

### DISCUSSION THREAD

**Discussion by mackelito** [simondotm/nx-firebase#187] [Q&A]
So I'm a rookie when it comes to firebase functions so I might be totally off here..
but reading over at https://firebase.google.com/docs/functions/config-env?gen=2nd#migrating_from_environment_configuration makes me think that env variables will be replaces by secrets!?

Also I wonder how I should think about secrets and config..?
Lets say I have apiKey, one for prod and one for test.. normally I would use the .env and .env.local.. but now I'm really not sure :D

from what I can tell.. deploying will include the .env file but it seems as firebase discourage from using this approach as it's not considered safe to have data like this in your repository...

**Comment by simondotm** [comment]
Hi @mackelito 
So environment var files are exactly that. They set variables that are visible either just in your local emulator suite (`.env.local`), or in your deployed function as well (`.env`).

Although you technically _can_ put secrets in them, this is not a good idea because as you rightly say, secrets should not exist in source code repositories.

If you really want to do it this way, you could add `.env` files to your `.gitignore` file for your workspace, and manage that file manually.


The recommended solutions to this are the firebase function configs (which is gen1 functions only and deprecated), or use `.secret` files - see https://firebase.google.com/docs/functions/config-env?gen=2nd#secret_parameters

**Comment by mackelito** [comment]
But let's say I want to use secrets.. should I have 2 secrets per environment? How do we check for environment in our main.ts file?
